src/nfts/NftsInfoCarousel.tsx

```diff
@@ -0,0 +1,369 @@
+import OpenLinkIcon from 'src/icons/OpenLinkIcon'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { Nft } from 'src/nfts/types'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import Logger from 'src/utils/Logger'
+import networkConfig from 'src/web3/networkConfig'
+
+const TAG = 'nfts/NftsInfoCarousel'
+const DEFAULT_IMAGE_HEIGHT = 360
+
+function scaleImageHeight(originalWidth: number, originalHeight: number, targetWidth: number) {
+  const aspectRatio = originalWidth / originalHeight
+  return targetWidth / aspectRatio
+}
+
+function NftsLoadErrorScreen() {
+  const { t } = useTranslation()
+  return (
+    <SafeAreaView
+      testID="NftsInfoCarousel/NftsLoadErrorScreen"
+      style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}
+    >
+      <Text>{t('nftInfoCarousel.noNftsFound')}</Text>
+    </SafeAreaView>
+  )
+}
+
+function ThumbnailImagePlaceholder() {
+  return (
+    <SkeletonPlaceholder backgroundColor={colors.gray2} highlightColor={colors.white}>
+      <View style={styles.skeletonImageThumbnailPlaceHolder} />
+    </SkeletonPlaceholder>
+  )
+}
+
+interface NftThumbnailProps {
+  nft: Nft
+  isActive: boolean
+  onPress: () => void
+}
+
+function NftThumbnail({ nft, isActive, onPress }: NftThumbnailProps) {
+  const [loading, setLoading] = useState(true)
+  return (
+    <Touchable
+      style={[
+        isActive ? styles.nftThumbnailSelected : styles.nftThumbnailUnselected,
+        !nft.metadata && styles.errorThumbnail,
+      ]}
+      borderless={false}
+      onPress={onPress}
+      testID={`NftsInfoCarousel/NftThumbnail/${nft.contractAddress}-${nft.tokenId}`}
+    >
+      {!nft.metadata ? (
+        <InfoIcon
+          size={isActive ? 24 : 20}
+          color={colors.warning}
+          testID="NftsInfoCarousel/ErrorIcon"
+        />
+      ) : (
+        <FastImage
+          style={[
+            styles.nftThumbnailShared,
+            isActive ? styles.nftThumbnailSelected : styles.nftThumbnailUnselected,
+          ]}
+          source={{
+            uri: nft.media.find((media) => media.raw === nft.metadata?.image)?.gateway,
+          }}
+          onError={() => {
+            Logger.error(
+              TAG,
+              `Error loading Nft thumbnail image for Nft contractAddress: ${nft.contractAddress} tokenId: ${nft.tokenId}`
+            )
+          }}
+          onLoadEnd={() => {
+            setLoading(false)
+          }}
+          resizeMode={FastImage.resizeMode.cover}
+        >
+          {loading && <ThumbnailImagePlaceholder />}
+        </FastImage>
+      )}
+    </Touchable>
+  )
+}
+interface MainImagePlaceholderProps {
+  height: number
+}
+
+function MainImagePlaceholder({ height }: MainImagePlaceholderProps) {
```

Comment:
nit: i think that this component could be combined with ThumbnailImagePlaceholder, where it defaults to the thumbnail height unless a height is passed in?

Reply:
Fixed in https://github.com/valora-inc/wallet/pull/3800/commits/6a0419157af35e7f6b0374818be5bcdb6c1e356d!